app/packages/core/src/components/ColorModal/ColorModal.tsx

```diff
@@ -41,27 +68,75 @@ const ColorModal = () => {
           aria-labelledby="draggable-color-modal"
         >
           <Draggable bounds="parent" handle=".draggable-colorModal-handle">
-            <Container>
-              <DraggableModalTitle className="draggable-colorModal-handle">
-                <div style={{ margin: "4px" }}>Edit color scheme</div>
-                <CloseIcon
-                  onClick={() => setActiveColorModalField(null)}
-                  onMouseDown={(e) => e.stopPropagation()}
-                  style={{ margin: "4px", cursor: "pointer" }}
-                />
-              </DraggableModalTitle>
-              <DraggableContent>
-                <SidebarList />
-                <Display>
-                  {field === ACTIVE_FIELD.global && <GlobalSetting />}
-                  {field === ACTIVE_FIELD.json && <JSONViewer />}
-                  {typeof field !== "string" && field && (
-                    <FieldSetting field={activeColorModalField as Field} />
-                  )}
-                </Display>
-              </DraggableContent>
-              <ColorFooter />
-            </Container>
+            <Resizable
+              size={{ height, width }}
+              minWidth={600}
+              minHeight={500}
+              enable={{
+                top: true,
+                right: true,
+                left: true,
+                bottom: true,
+                topRight: true,
+                bottomRight: true,
+                bottomLeft: true,
+                topLeft: true,
+              }}
+              onResizeStop={(e, direction, ref, { width: dw, height: dh }) => {
+                setWidth(width + dw);
+                setHeight(height + dh);
+                // reset sidebar width on double click
+                if (e.detail === 2) {
+                  setWidth(860);
+                  setHeight(680);
+                }
+              }}
+              handleStyles={{
+                ["right"]: { right: 0, width: 4 },
```

Comment:
suggestion for future (may ignore): we should move these to a common util. I think it's duplicated in couple placed